libraries/AP_HAL_ChibiOS/hwdef/quanton/hwdef-bl.dat

```diff
@@ -0,0 +1,45 @@
+# hw definition file for processing by chibios_pins.py
+# for minimal F405 bootloader
+
+# MCU class and specific type
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 9999
```

Comment:
This is very much Not Right.  You need to allocate a board ID in `Tools/AP_Bootloader/board_types.txt` and use it here.

Reply:
Hi Peter ! Thank you very much for your comment. Shall I use the same ID 9999 or you can allocate new and correct number for me ?

Reply:
@magicrub According hardware board is modern enough and I am not sure if it still is available, but I saw in the forums some people tried to find Ardupilot solution for this board.

Reply:
It’s an older board, i have one that I can test on. It’s virtually the same pins as the TBS colibri F4

Reply:
> It’s an older board, i have one that I can test on. It’s virtually the same pins as the TBS colibri F4
> Also it’s 16Mhz oscillator

Yes, I added this to the PR in the comments hwdef.dat:
# Original Quanton 16MHz crystall have to be changed to 8MHz !!!
# crystal frequency

Reply:
Are you saying that users have to change their oscillators to 8mhz (hardware) ?

Reply:
At least I find only this way how to use this board with Ardupilot. For me it was very easy.

Reply:
`# ------Frequency-------------
# crystal frequency
OSCILLATOR_HZ 16000000
define HAL_CUSTOM_CLOCK_TREE
define STM32_PLLM_VALUE 8
define STM32_PLLN_VALUE 168
define STM32_PLLP_VALUE 2
define STM32_PLLQ_VALUE 7`

You could add this.

Reply:
My knowledges are not enough to understand if such kind of change will work with the whole Chibios and Ardupilot infrastructure

Reply:
I can test this for you, as this will allow the user not to have to do hardware changes to a board that may well already be mounted in a vehicle and also with the lack of skills to undertake the change might damage the flight controller beyond repair,

Reply:
I agree if this will work for sure it is much easy.

Reply:
Just tested with custom clock changes and its working also changed the LED's as without them flashing i thought the boards was dead, [Quanton hwdef](https://github.com/ukmook/ardupilot/blob/QCopter-4.1/libraries/AP_HAL_ChibiOS/hwdef/Quanton/hwdef.dat)

Reply:
with board I have LED ports are correct.

Reply:
I have not changed pin assignment, just made them flash. They are still PC13 PC14

Reply:
I see, thanks for the changes.